apps/mobile-wallet/src/contexts/walletConnect/WalletConnectContext.tsx

```diff
@@ -304,373 +302,289 @@ export const WalletConnectContextProvider = ({ children }: { children: ReactNode
+              setLoading('Approving...')
+
+              try {
+                const signResult = await sendTransaction()
+
+                if (!signResult) {
+                  console.log('⏳ DID NOT GET A SIGNATURE RESULT, INFORMING DAPP THAT SESSION REQUEST FAILED...')
+                  await respondToWalletConnectWithError(sessionRequestEvent, {
+                    message: 'Sending transaction failed',
+                    code: WALLETCONNECT_ERRORS.TRANSACTION_SEND_FAILED
+                  })
+                  console.log('✅ INFORMING: DONE!')
+                } else {
+                  console.log('⏳ INFORMING DAPP THAT SESSION REQUEST SUCCEEDED...')
+                  await respondToWalletConnectWithSuccess(sessionRequestEvent, signResult)
+                  console.log('✅ INFORMING: DONE!')
+                }
+              } catch (e) {
+                console.error('❌ INFORMING: FAILED.')
+              } finally {
+                console.log('👉 RESETTING SESSION REQUEST EVENT.')
+                setSessionRequestEvent(undefined)
+                setSessionRequestData(undefined)
+                setLoading('')
+                showToast({ text1: t('dApp request approved'), text2: t('You can go back to your browser.') })
+              }
+            }
+          })
+      })
+    },
+    [
+      respondToWalletConnectWithError,
+      respondToWalletConnectWithSuccess,
+      sessionRequestEvent,
+      t,
+      triggerBiometricsAuthGuard,
+      triggerFundPasswordAuthGuard
+    ]
+  )
+
+  const openWalletConnectSessionProposalModal = useCallback(
+    () =>
+      sessionProposalEvent &&
+      dispatch(
+        openModal({
+          name: 'WalletConnectSessionProposalModal',
+          props: {
+            approveProposal,
+            rejectProposal,
+            proposalEvent: sessionProposalEvent
           }
-          case 'alph_signUnsignedTx': {
-            const { signerAddress, signerKeyType, unsignedTx } = requestEvent.params.request
-              .params as SignUnsignedTxParams
+        })
+      ),
+    [approveProposal, dispatch, rejectProposal, sessionProposalEvent]
+  )
 
-            const fromAddress = addressIds.find((address) => address === signerAddress)
+  const onSessionProposal = useCallback(
+    async (sessionProposalEvent: SessionProposalEvent) => {
+      console.log('📣 RECEIVED EVENT PROPOSAL TO CONNECT TO A DAPP!')
+      console.log('👉 ARGS:', sessionProposalEvent)
+      console.log('⏳ WAITING FOR PROPOSAL APPROVAL OR REJECTION')
 
-            if (!fromAddress) {
-              return respondToWalletConnectWithError(requestEvent, {
-                message: "Signer address doesn't exist",
-                code: WALLETCONNECT_ERRORS.SIGNER_ADDRESS_DOESNT_EXIST
-              })
-            }
+      setSessionProposalEvent(sessionProposalEvent)
+      openWalletConnectSessionProposalModal()
+    },
+    [openWalletConnectSessionProposalModal]
+  )
```

Comment:
This doesn't work.

```
      setSessionProposalEvent(sessionProposalEvent)
      openWalletConnectSessionProposalModal()
```

because in the `openWalletConnectSessionProposalModal` function, `sessionProposalEvent` has not yet received a value before it's called.

Reply:
Same problem with `openWalletConnectSessionRequestModal`.